contracts/src/systems/realm/contracts.cairo

```diff
@@ -242,6 +274,16 @@ mod realm_systems {
 
             quest.completed = true;
             world.write_model(@quest);
+
```

Comment:
could you add a check after `let quest_reward_config: QuestRewardConfig = world.read_model(quest_id);` to assert that `quest_reward_config.detached_resource_count > 0`. so we're sure that the last quest is really the last one. they cant just put an arbitrary `quest_id`